ibis/backends/tests/test_generic.py

```diff
@@ -995,3 +995,93 @@ def test_pivot_longer(backend):
     )
     df = res.limit(5).execute()
     assert not df.empty
+
+
+@pytest.mark.parametrize(
+    "on",
```

Comment:
See my comment below with merging this with the existing https://github.com/ibis-project/ibis/blob/9bcdf77ddafce75f0e5d8714d01dde81ed0b90f2/ibis/tests/sql/test_sqlalchemy.py#L358

That existing function implicitly tests the cases of on=None and keep=None, but perhaps you want to merge/update the tests so you explicitly are testing those cases as well?

Reply:
I'd rather do that in a follow-up so that there's less code to review in this PR.

Reply:
These are different kinds of tests, testing different things. I'm going to leave them as is.